envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/groups/NodeMetadata.kt

```diff
@@ -487,27 +507,30 @@ data class Outgoing(
     private val serviceDependencies: List<ServiceDependency> = emptyList(),
     private val domainDependencies: List<DomainDependency> = emptyList(),
     private val domainPatternDependencies: List<DomainPatternDependency> = emptyList(),
+    private val tagDependencies: List<TagDependency> = emptyList(),
     val allServicesDependencies: Boolean = false,
     val defaultServiceSettings: DependencySettings = DependencySettings()
 ) {
 
     // not declared in primary constructor to exclude from equals(), copy(), etc.
-    private val deduplicatedDomainDependencies: List<DomainDependency> = domainDependencies
-        .map { it.domain to it }
-        .toMap().values.toList()
+    private val deduplicatedDomainDependencies: List<DomainDependency> =
+        domainDependencies.associateBy { it.domain }.values.toList()
+
+    private val deduplicatedServiceDependencies: List<ServiceDependency> =
+        serviceDependencies.associateBy { it.service }.values.toList()
 
-    private val deduplicatedServiceDependencies: List<ServiceDependency> = serviceDependencies
-        .map { it.service to it }
-        .toMap().values.toList()
+    private val deduplicatedDomainPatternDependencies: List<DomainPatternDependency> =
+        domainPatternDependencies.associateBy { it.domainPattern }.values.toList()
 
-    private val deduplicatedDomainPatternDependencies: List<DomainPatternDependency> = domainPatternDependencies
-        .map { it.domainPattern to it }
-        .toMap().values.toList()
+    private val deduplicatedTagDependency: List<TagDependency> =
+        tagDependencies.associateBy { it.tag }.values.toList()
```

Comment:
In the previous implementation last key occurrence was taken to the final collection. So I used `associateBy` because it has the same logic. I believe `distinctBy` has different logic and take the first occurrence of key